### PR TITLE
Use fresh null label and context

### PR DESCRIPTION
**Pull request by okgolove**
## what
* Use context feature and latest null-label module

## why
* Some settings like delimiter didn't work for this module

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -18,7 +18,7 @@ locals {
 }
 
 module "base_label" {
-  source              = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.17.0"
+  source              = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.19.2"
   namespace           = var.namespace
```
Comment (aknysh) [inline main.tf:22]: remove this label, we have it in `context.tf`

File: main.tf [hunk 2]
```diff
@@ -33,7 +33,7 @@ module "base_label" {
 }
 
 module "s3_bucket_label" {
-  source  = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.17.0"
+  source  = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.19.2"
   context = module.base_label.context
```
Comment (aknysh) [inline main.tf:37]:
```suggestion
  context = module.this.context
```

`module.this` is the label from `context.tf`

File: main.tf [hunk 3]
```diff
@@ -172,7 +172,7 @@ resource "aws_s3_bucket_public_access_block" "default" {
 }
 
 module "dynamodb_table_label" {
-  source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.17.0"
+  source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.19.2"
   context    = module.base_label.context
```
Comment (aknysh) [inline main.tf:176]:
```suggestion
  context    = module.this.context
```

File: main.tf (hunk 1 above)
Comment (aknysh) [reply main.tf:22]: replace all occurrences of `module.base_label.xxx` with `module.this.xxx`

**Review by aknysh — COMMENTED** [review]
thanks @okgolove 
LGTM, see comments

**Comment by okgolove** [comment]
@aknysh thank you for the review!
I've updated the PR.

**Comment by aknysh** [comment]
@okgolove 

Thanks for the updates, looks good.

Please fix this error 

```
TestExamplesComplete 2020-11-09T18:02:11Z command.go:121: Error: Reference to undeclared module
TestExamplesComplete 2020-11-09T18:02:11Z command.go:121: 
TestExamplesComplete 2020-11-09T18:02:11Z command.go:121:   on ../../replication.tf line 4, in resource "aws_iam_role" "replication":
TestExamplesComplete 2020-11-09T18:02:11Z command.go:121:    4:   name               = format("%s-replication", module.base_label.id)
TestExamplesComplete 2020-11-09T18:02:11Z command.go:121: 
TestExamplesComplete 2020-11-09T18:02:11Z command.go:121: No module call named "base_label" is declared in tfstate_backend.
TestExamplesComplete 2020-11-09T18:02:11Z command.go:121: 

```

**Comment by okgolove** [comment]
Oops. 
Done :)

File: context.tf [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+# for example `context = module.this.context`,
+# and access individual variables as `module.this.<var>`,
+# with final values filled in.
+#
+# For example, when using defaults, `module.this.context.delimiter`
+# will be null, and `module.this.delimiter` will be `-` (hyphen).
+#
+
+module "this" {
+  source = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.19.2"
```
Comment (aknysh) [inline context.tf:22]:
```suggestion
  source = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.21.0"
```

File: main.tf [hunk 5]
```diff
@@ -17,24 +17,9 @@ locals {
...
-  additional_tag_map  = var.additional_tag_map
-  context             = var.context
-  label_order         = var.label_order
-  regex_replace_chars = var.regex_replace_chars
-}
-
 module "s3_bucket_label" {
-  source  = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.17.0"
-  context = module.base_label.context
+  source  = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.19.2"
```
Comment (aknysh) [inline main.tf:21]:
```suggestion
  source  = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.21.0"
```

File: main.tf [hunk 6]
```diff
@@ -172,8 +157,8 @@ resource "aws_s3_bucket_public_access_block" "default" {
 }
 
 module "dynamodb_table_label" {
-  source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.17.0"
-  context    = module.base_label.context
+  source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.19.2"
```
Comment (aknysh) [inline main.tf:160]:
```suggestion
  source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.21.0"
```

**Review by aknysh — COMMENTED** [review]
@okgolove LGTM
sorry, we need to update the label modules to `0.21.0` to be able to support TF 0.14.
also, please run the following commands from the repo root:

```
make init
make github/init
make readme
```

thank you

**Comment by aknysh** [comment]
@okgolove 
can you please fix the terratest in https://github.com/cloudposse/terraform-aws-tfstate-backend/blob/master/test/src/examples_complete_test.go

thanks again


```
--- FAIL: TestExamplesComplete (36.98s)
    examples_complete_test.go:33: 
        	Error Trace:	examples_complete_test.go:33
        	Error:      	Not equal: 
        	            	expected: "eg-test-terraform-tfstate-backend-state"
        	            	actual  : "eg-test-terraform-tfstate-backend"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-eg-test-terraform-tfstate-backend-state
        	            	+eg-test-terraform-tfstate-backend
        	Test:       	TestExamplesComplete
    examples_complete_test.go:40: 
        	Error Trace:	examples_complete_test.go:40
        	Error:      	Not equal: 
        	            	expected: "eg-test-terraform-tfstate-backend-state-lock"
        	            	actual  : "eg-test-terraform-tfstate-backend-lock"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-eg-test-terraform-tfstate-backend-state-lock
        	            	+eg-test-terraform-tfstate-backend-lock
        	Test:       	TestExamplesComplete
```

**Comment by okgolove** [comment]
@aknysh it used to be a default `attributes` variable.
Would it be ok if I hardcode this?

```
module "s3_bucket_label" {
  source = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.21.0"

  attributes = compact(concat(var.attributes, ["state"]))

  context = module.this.context
}

module "dynamodb_table_label" {
  source = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.21.0"

  attributes = compact(concat(var.attributes, ["state", "lock"]))

  context = module.this.context
}
```

**Comment by aknysh** [comment]
> @aknysh it used to be a default `attributes` variable.
> Would it be ok if I hardcoded this?
> 
> ```
> module "s3_bucket_label" {
>   source = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.21.0"
> 
>   attributes = compact(concat(var.attributes, ["state"]))
> 
>   context = module.this.context
> }
> 
> module "dynamodb_table_label" {
>   source = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.21.0"
> 
>   attributes = compact(concat(var.attributes, ["state", "lock"]))
> 
>   context = module.this.context
> }
> ```

you can just update the Go code in the terratest (no need to change the module)

**Comment by okgolove** [comment]
I got you wrong. I've fixed the test.

**Review by aknysh — APPROVED** [review]
thanks @okgolove

**Pull request merged**